app/Jobs/ImportTurboVotePosts.php

```diff
@@ -340,5 +320,63 @@ private function scrubRecord($record)
 
         return $isNotValidEmail || $isNotValidHostname || $isNotValidLastName ? false : true;
     }
+
+    /**
+     * For a given record and referral code values, first check if we have a northstar ID, then grab the user using that.
+     * Otherwise, see if we can find the user with the given email (if it exists), if not check if we can find them with the given phone number (if it exists).
+     * If all else fails, create the user .
+     *
+     * @TODO - If we have a northstar id in the referral code, then we probably don't need to make a call to northstar for the full user object.
+     *
+     * @return array
+     */
+    private function getOrCreateUser($record, $values)
```

Comment:
I think that this method makes sense. Nothing really jumps out at me as being unclear either!